src/transform-sdk/rust/core-types/src/lib.rs

```diff
@@ -116,13 +116,36 @@ impl<'a> From<&'a WrittenRecord<'a>> for BorrowedRecord<'a> {
     }
 }
 
+/// [`WriteOptions`] allows for customizing a [`RecordWriter`]'s write.
+///
+/// An example is to customize the output topic which a write can go to.
+#[derive(Default, PartialEq, Eq, Clone, Debug)]
+pub struct WriteOptions<'a> {
+    pub topic: Option<&'a str>,
+}
+
+impl<'a> WriteOptions<'a> {
+    /// Create a new options struct with the record destination to an optional `topic`.
+    ///
+    /// If `topic` is `None` then the record will be written to the first
+    /// output topic in listed in the configuration.
+    pub fn new(topic: Option<&'a str>) -> Self {
+        Self { topic }
+    }
+
+    /// Create a new options struct with the record destination to `topic`.
```

Comment:
Same question as above regarding record/`Record`

Reply:
Yes, updated it, PTAL